pipeline.py

```diff
@@ -295,6 +323,22 @@ def _get_endpoint(pipeline_job):
     pipeline_job.task_details)
   raise RuntimeError("Unexpected deploy result format")
 
+def _gpu_to_dsm(gpu_type):
+  if gpu_type.lower() == "NVIDIA-TESLA-P4".lower():
```

Comment:
nit: seems like we can use .upper() on the left part and ommit the .lower on the right part